src/sqlparser/src/parser.rs

```diff
@@ -2780,10 +2781,10 @@ impl Parser {
         } else {
             let name = self.parse_object_name()?;
             // Postgres,table-valued functions:
-            let args = if self.consume_token(&Token::LParen) {
+            let (args, _) = if self.consume_token(&Token::LParen) {
```

Comment:
Either report an error here when `order_by` is not empty, or return it as part of `TableFactor::Table` and let the upper level decide whether it is valid.